Orange/widgets/data/owselectrows.py

```diff
@@ -628,20 +626,33 @@ def commit(self):
         self.match_desc = report.describe_data_brief(matching_output)
         self.nonmatch_desc = report.describe_data_brief(non_matching_output)
 
-        self.update_info(matching_output, self.data_out_rows, "Out: ")
+        self.update_output_summary(matching_output)
 
-    def update_info(self, data, lab1, label):
+    def update_input_summary(self, data):
         def sp(s, capitalize=True):
             return s and s or ("No" if capitalize else "no"), "s" * (s != 1)
 
-        if data is None:
-            lab1.setText("")
+        if data:
+            n = data.approx_len()
+            details = "~%s row%s, %s variable%s" % \
+                      (sp(n) + sp(len(data.domain.variables) + len(data.domain.metas)))
+            n = self.info.format_number(n)
+            self.info.set_input_summary(n, details)
         else:
-            lab1.setText(label + "~%s row%s, %s variable%s" %
-                         (sp(data.approx_len()) +
-                          sp(len(data.domain.variables) +
-                             len(data.domain.metas)))
-                        )
+            self.info.set_input_summary(self.info.NoInput)
+
+    def update_output_summary(self, data):
+        def sp(s, capitalize=True):
```

Comment:
This function is copied from `update_input_summary`. Consider creating a staticmethod to avoid the duplicated code.

EDIT: Actually, the majority of the code in this method is duplicated. You could replace it with something like:

```     
    def _create_summary(self, data):
        def sp(s, capitalize=True):
            return s and s or ("No" if capitalize else "no"), "s" * (s != 1)

        summary, details = self.info.NoInput, ""
        if data:
            n = data.approx_len()
            details = "~%s row%s, %s variable%s" % \
                      (sp(n) + sp(len(data.domain.variables) +
                                  len(data.domain.metas)))
            summary = self.info.format_number(n)
        return summary, details
``` 

and then call it from `update_input_summary` and `update_output_summary`:
`self.info.set_input_summary(*self._create_summary(data))`